src/plugins/tabs/components/tabs.vue

```diff
@@ -135,15 +163,23 @@ export default {
         },
         removeItem(identifier) {
             let pos = this.tabsList.findIndex(tab =>
-                    tab.domainObject.identifier.namespace === identifier.namespace && tab.domainObject.identifier.key === identifier.key
-                ),
-                tabToBeRemoved = this.tabsList[pos];
+                tab.domainObject.identifier.namespace === identifier.namespace && tab.domainObject.identifier.key === identifier.key
+            );
+
+            if(pos < 0) {
+                return;
+            }
 
+            let tabToBeRemoved = this.tabsList[pos];
             this.tabsList.splice(pos, 1);
 
             if (this.isCurrent(tabToBeRemoved)) {
                 this.showTab(this.tabsList[this.tabsList.length - 1], this.tabsList.length - 1);
             }
+
+            this.openmct.objects.get(identifier).then((childDomainObject) => {
+                this.RemoveAction.removeFromComposition(this.internalDomainObject, childDomainObject);
```

Comment:
Instead of importing the remove Action here, we can just call removeItem from the click handler, and we can add a flag in removeAction to say removeFromComposition, then we can call this.composition.remove(this.domainObject, identifierToBeRemoved) if the flag (removeFromComposition) is true.

Reply:
Openmct.objects.get adds overhead that we should not need when removing an item